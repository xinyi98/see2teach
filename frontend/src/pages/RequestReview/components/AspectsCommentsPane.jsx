import { Tabs, Input } from 'antd';

const { TabPane } = Tabs;
const { TextArea } = Input;

// show the details of selected aspects
const AspectsCommentsPane = ({ selectedAspects, dispatch }) => {
  const updateComment = (aspect_name, reviewee_comments) => {
    dispatch({ type: 'aspectComments', aspect_name, reviewee_comments });
  };

  return (
    <Tabs type="card" size="small" tabBarStyle={{ marginBottom: 0 }}>
      {selectedAspects.map((aspect, key) => (
        <TabPane tab={aspect.aspect_name} key={key}>
          <TextArea
            placeholder={`${aspect.aspect_name} comments...`}
            value={aspect.reviewee_comments}
            rows={6}
            onChange={(e) => updateComment(aspect.aspect_name, e.target.value)}
          />
        </TabPane>
      ))}
    </Tabs>
  );
};

export default AspectsCommentsPane;
