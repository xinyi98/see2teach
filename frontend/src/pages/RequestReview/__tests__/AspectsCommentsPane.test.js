import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AspectsCommentsPane from '../components/AspectsCommentsPane';
import '@testing-library/jest-dom';

afterEach(() => {
  cleanup();
});

test('snapshot test for aspects comments pane', () => {
  const aspectComments = ['Pace', 'General'];
  const dispatch = jest.fn();
  const tree = renderer
    .create(<AspectsCommentsPane selectedAspects={aspectComments} dispatch={dispatch} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
