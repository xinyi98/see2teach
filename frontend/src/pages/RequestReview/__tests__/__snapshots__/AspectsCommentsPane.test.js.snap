// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for aspects comments pane 1`] = `
<div
  className="ant-tabs ant-tabs-top ant-tabs-small ant-tabs-card"
>
  <div
    className="ant-tabs-nav"
    onKeyDown={[Function]}
    role="tablist"
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    <div
      className="ant-tabs-nav-wrap"
    >
      <div
        className="ant-tabs-nav-list"
        style={
          Object {
            "transform": "translate(0px, 0px)",
            "transition": undefined,
          }
        }
      >
        <div
          className="ant-tabs-tab ant-tabs-tab-active"
          onClick={[Function]}
        >
          <div
            aria-controls={null}
            aria-selected={true}
            className="ant-tabs-tab-btn"
            id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={0}
          />
        </div>
        <div
          className="ant-tabs-tab"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": undefined,
            }
          }
        >
          <div
            aria-controls={null}
            aria-selected={false}
            className="ant-tabs-tab-btn"
            id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={0}
          />
        </div>
        <div
          className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
        />
      </div>
    </div>
    <div
      className="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      <button
        aria-controls="null-more-popup"
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-hidden="true"
        className="ant-tabs-nav-more"
        id="null-more"
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "marginLeft": undefined,
            "order": 1,
            "visibility": "hidden",
          }
        }
        tabIndex={-1}
        type="button"
      >
        <span
          aria-label="ellipsis"
          className="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className="ant-tabs-content-holder"
  >
    <div
      className="ant-tabs-content ant-tabs-content-top"
      style={null}
    >
      <div
        aria-hidden={false}
        aria-labelledby={null}
        className="ant-tabs-tabpane ant-tabs-tabpane-active"
        id={null}
        role="tabpanel"
        style={Object {}}
        tabIndex={0}
      >
        <textarea
          className="ant-input"
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onKeyDown={[Function]}
          placeholder="undefined comments..."
          rows={6}
          style={Object {}}
          value=""
        />
      </div>
      <div
        aria-hidden={true}
        aria-labelledby={null}
        className="ant-tabs-tabpane"
        id={null}
        role="tabpanel"
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex={-1}
      />
    </div>
  </div>
</div>
`;
